### PR TITLE
#102 Provide endpoint that generates qr code for a digital eak

### LINKED ISSUE #102: Provide endpoint that generates qr code for a digital eak
Refers to #83 

- [ ] Endpoint in the backend can be reached via http
- [ ] Caller provides eak owners first name, last name, region (landkreis) and expiration date
- [ ] Generate hash from the provided data with an additional random string
- [ ] Store the hash in the database, maybe also store creation timestamp and expiration date
- [ ] Generate a shared OTP secret and store it in the database
- [ ] Encode firstname, lastname, region, expiration date, random string and otp secret in a QR code
- [ ] A QR code is sent back in the reply, either pdf or png

### PR DESCRIPTION
**Pull request by RoAnBu**
closes #102

### CODE REVIEW

File: .idea/jarRepositories.xml [hunk 1]
```diff
@@ -21,5 +21,10 @@
       <option name="name" value="MavenRepo" />
       <option name="url" value="https://repo.maven.apache.org/maven2/" />
     </remote-repository>
+    <remote-repository>
+      <option name="id" value="maven" />
+      <option name="name" value="maven" />
```
Comment (aiquita) [inline .idea/jarRepositories.xml:26]: Warum führst du jetzt maven ein?

File: backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/administration/card/CardDetails.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package xyz.elitese.ehrenamtskarte.administration.card
+
+import java.time.LocalDate
+
+data class CardDetails(val firstName: String, val lastName: String, val date: LocalDate)
```
Comment (aiquita) [inline backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/administration/card/CardDetails.kt]: Is this the expiration date?

File: backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/administration/serialization/QRCodeGenerator.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package xyz.elitese.ehrenamtskarte.administration.serialization
+
+import net.glxn.qrgen.javase.QRCode
+
+class QRCodeGenerator {
```
Comment (aiquita) [inline backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/administration/serialization/QRCodeGenerator.kt]: Please use object for mere singleton or functional stuff, respectively

File: backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/administration/webservice/AdminWebService.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            cfg.enableDevLogging()
+            cfg.enableCorsForAllOrigins()
+            cfg.registerPlugin(OpenApiPlugin(getOpenApiOptions()))
+        }.start(PORT)
+
+        println("Admin Server is running at http://localhost:${PORT}")
+
+        val cardIssueHandler = CardIssueHandler()
+
+        app.get("/card", documented(CardIssueHandler.getCardDocumentation) {
```
Comment (aiquita) [inline backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/administration/webservice/AdminWebService.kt]: Is there a more precise route?
For example /issueCard

File: .idea/jarRepositories.xml (hunk 1 above)
Comment (RoAnBu) [reply .idea/jarRepositories.xml:26]: Gradle uses maven repos to pull dependencies, in this case a custom repo. We still dont use the program maven though :D

File: backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/administration/card/CardDetails.kt (hunk 2 above)
Comment (RoAnBu) [reply backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/administration/card/CardDetails.kt]: yes fixed in newer commit

File: backend/src/main/kotlin/app/ehrenamtskarte/administration/card/CardIssuer.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import app.ehrenamtskarte.administration.serialization.QRCodeGenerator
+import com.beust.klaxon.Klaxon
+import io.jsonwebtoken.SignatureAlgorithm
+import io.jsonwebtoken.security.Keys
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+
+class CardIssuer(
+    private val cardTokenGenerator: CardTokenGenerator =
```
Comment (aiquita) [inline backend/src/main/kotlin/app/ehrenamtskarte/administration/card/CardIssuer.kt:12]: In which cases could it be necessary to exchange this?

**Review by aiquita — CHANGES_REQUESTED** [review]

File: backend/src/main/kotlin/app/ehrenamtskarte/administration/serialization/QRCodeGenerator.kt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package app.ehrenamtskarte.administration.serialization
+
+import net.glxn.qrgen.javase.QRCode
+
+class QRCodeGenerator {
```
Comment (aiquita) [inline backend/src/main/kotlin/app/ehrenamtskarte/administration/serialization/QRCodeGenerator.kt:5]: Please use objects for static or functional stuff, respectively

File: backend/src/main/kotlin/app/ehrenamtskarte/administration/webservice/AdminWebService.kt [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            cfg.enableDevLogging()
+            cfg.enableCorsForAllOrigins()
+            cfg.registerPlugin(OpenApiPlugin(getOpenApiOptions()))
+        }.start(PORT)
+
+        logger.info("Admin Server is running at http://localhost:${PORT}")
+
+        val cardIssueHandler = CardIssueHandler()
+
+        app.get("/card", documented(CardIssueHandler.getCardDocumentation) {
```
Comment (aiquita) [inline backend/src/main/kotlin/app/ehrenamtskarte/administration/webservice/AdminWebService.kt:29]: Is there a more precise route?

/issueCard for example

**Review by aiquita — CHANGES_REQUESTED** [review]

File: backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/webservice/WebService.kt [hunk 8]
```diff
@@ -13,7 +13,7 @@ class WebService {
             cfg.addStaticFiles("/graphiql", "/graphiql", Location.CLASSPATH)
         }.start(PORT)
 
-        println("Server is running at http://localhost:7000")
+        println("Server is running at http://localhost:${PORT}")
         println("Goto http://localhost:7000/graphiql for a graphical editor")
```
Comment (Schedulaar) [inline backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/webservice/WebService.kt:17]:
```suggestion
        println("Goto http://localhost:${PORT}/graphiql for a graphical editor")
```

**Comment by RoAnBu** [comment]
Will be replaced by new PR

**Pull request closed without merging**